# ----------------------------------------------------------------------------------------------
#   Request
# ----------------------------------------------------------------------------------------------

# --- Different modes for controlling linear velocity [ x , y , z ] ---
int32 target_mode

# [ target_mode = 0 ]  (navigate through pose(s))
# Robot traverses through each pose with respect to its reference frame.
geometry_msgs/PoseStamped[] poses

# [ target_mode = 1 ]  (navigate through waypoint(s))
# Robot traverses to the waypoint specified by the PoseStamped msg on the waypoint topic, relative to world.
string waypoint_topic


# --- Different modes for controlling angular velocity [ roll , pitch , yaw ] ---
int32 orient_mode

# [ orient_mode = 0 ]  (orient relative to path)
# Orient along the path of travel. Offset relative to direction of travel (ie “forward facing path”: offset=[0,0,0])
geometry_msgs/Quaternion offset_orientation

# [ orient_mode = 1 ]  (orient to constant)
# Orient as an explicit orientation during movement. Relative to world frame.
geometry_msgs/Quaternion constant_orientation

# [ orient_mode = 2 ]  (orient as next target)
# Orient as the next pose / waypoint target orientation. Orientations interpolated between targets.


# General parameters
string robot_frame          # the frame to move the robot through, must be base_link_ned or child of
bool level_lock             # lock roll and pitch to 0 during entire action
bool end_as_final_target    # irregardless of orient_mode, hold as the final target's orientation after execution

bool depth_lock             # NOT USED use depth value below during entire action
float32 depth               # NOT USED
float32 speed               # NOT USED [0, 1] how fast the robot should be moving
bool steady_target          # NOT USED make sure the robot is steady at the target for at least 1 second before finishing action
float32 position_tolerance  # NOT USED maximum distance delta for target to be considered reached (0 for default)
float32 velocity_tolerance  # NOT USED maximum velocity for the robot to be considered steady (0 for default)

---

# ----------------------------------------------------------------------------------------------
#   Result
# ----------------------------------------------------------------------------------------------

bool success

---

# ----------------------------------------------------------------------------------------------
#   Feedback
# ----------------------------------------------------------------------------------------------

float32 travelled
float32 remaining
float32 linear_speed
float32 angular_speed
